server/src/services/etablissement.service.ts

```diff
@@ -209,6 +210,25 @@ export const getGeoPoint = async (adresse: string): Promise<IGeoPoint> => {
   }
 }
 
+export const getGeoFeature = async (adresse: string): Promise<IFeature> => {
```

Comment:
le typage de IFeature n'est pas correct (cf https://raw.githubusercontent.com/geocoders/geocodejson-spec/refs/heads/master/draft/geocodejson.schema.json)

Reply:
transformé en typescript, ça donne ça :

```
/**
 * GeocodeJSON is an extension of the GeoJSON format and it is an attempt to create a standard for handling geocoding results.
 */
export type GeocodeJSONSchema = GeoJSON & {
  /**
   * REQUIRED. GeocodeJSON result is a FeatureCollection.
   */
  type?: "FeatureCollection"
  /**
   * REQUIRED. Namespace.
   */
  geocoding: {
    /**
     * A semver.org compliant version number. Describes the version of the GeocodeJSON spec that is implemented by this instance.
     */
    version: string
    /**
     * OPTIONAL. The licence of the data. In case of multiple sources, and then multiple licences, can be an object with one key by source.
     */
    licence?: string
    /**
     * OPTIONAL. The attribution of the data. In case of multiple sources, and then multiple attributions, can be an object with one key by source.
     */
    attribution?: string
    /**
     * OPTIONAL. The query that has been issued to trigger the search.
     */
    query?: string
    [k: string]: unknown
  }
  [k: string]: unknown
} & {
  /**
   * REQUIRED. As per GeoJSON spec.
   */
  features?: {
    /**
     * REQUIRED. As per GeoJSON spec.
     */
    type?: "Feature"
    /**
     * REQUIRED. As per GeoJSON spec.
     */
    properties?: {
      /**
       * REQUIRED. Namespace.
       */
      geocoding: {
        /**
         * REQUIRED. One of house, street, locality, city, region, country.
         */
        type: string
        /**
         * OPTIONAL. Result accuracy, in meters.
         */
        accuracy?: number
        /**
         * RECOMMENDED. Suggested label for the result.
         */
        label?: string
        /**
         * OPTIONAL. Name of the place.
         */
        name?: string
        /**
         * OPTIONAL. Housenumber of the place.
         */
        housenumber?: string
        /**
         * OPTIONAL. Street of the place.
         */
        street?: string
        /**
         * OPTIONAL. Locality of the place.
         */
        locality?: string
        /**
         * OPTIONAL. Postcode of the place.
         */
        postcode?: string
        /**
         * OPTIONAL. City of the place.
         */
        city?: string
        /**
         * OPTIONAL. District of the place.
         */
        district?: string
        /**
         * OPTIONAL. County of the place.
         */
        county?: string
        /**
         * OPTIONAL. State of the place.
         */
        state?: string
        /**
         * OPTIONAL. Country of the place.
         */
        country?: string
        /**
         * OPTIONAL. Administratives boundaries the feature is included in, as defined in http://wiki.osm.org/wiki/Key:admin_level#admin_level.
         */
        admin?: {
          /**
           * This interface was referenced by `undefined`'s JSON-Schema definition
           * via the `patternProperty` "^level[0-9]+$".
           */
          [k: string]: string
        }
        /**
         * OPTIONAL. Geohash encoding of coordinates (see http://geohash.org/site/tips.html).
         */
        geohash?: string
        [k: string]: unknown
      }
      [k: string]: unknown
    }
    [k: string]: unknown
  }[]
  [k: string]: unknown
}
export type GeoJSON =
  | GeoJSONPoint
  | GeoJSONLineString
  | GeoJSONPolygon
  | GeoJSONMultiPoint
  | GeoJSONMultiLineString
  | GeoJSONMultiPolygon
  | GeoJSONGeometryCollection
  | GeoJSONFeature
  | GeoJSONFeatureCollection

export interface GeoJSONPoint {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONGeometryCollection {
  type: "GeometryCollection"
  geometries: (
    | GeoJSONPoint1
    | GeoJSONLineString1
    | GeoJSONPolygon1
    | GeoJSONMultiPoint1
    | GeoJSONMultiLineString1
    | GeoJSONMultiPolygon1
  )[]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPoint1 {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString1 {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon1 {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint1 {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString1 {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon1 {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONFeature {
  type: "Feature"
  id?: number | string
  properties: null | {
    [k: string]: unknown
  }
  geometry:
    | null
    | GeoJSONPoint2
    | GeoJSONLineString2
    | GeoJSONPolygon2
    | GeoJSONMultiPoint2
    | GeoJSONMultiLineString2
    | GeoJSONMultiPolygon2
    | GeoJSONGeometryCollection1
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPoint2 {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString2 {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon2 {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint2 {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString2 {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon2 {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONGeometryCollection1 {
  type: "GeometryCollection"
  geometries: (
    | GeoJSONPoint3
    | GeoJSONLineString3
    | GeoJSONPolygon3
    | GeoJSONMultiPoint3
    | GeoJSONMultiLineString3
    | GeoJSONMultiPolygon3
  )[]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPoint3 {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString3 {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon3 {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint3 {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString3 {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon3 {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONFeatureCollection {
  type: "FeatureCollection"
  features: GeoJSONFeature1[]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONFeature1 {
  type: "Feature"
  id?: number | string
  properties: null | {
    [k: string]: unknown
  }
  geometry:
    | null
    | GeoJSONPoint4
    | GeoJSONLineString4
    | GeoJSONPolygon4
    | GeoJSONMultiPoint4
    | GeoJSONMultiLineString4
    | GeoJSONMultiPolygon4
    | GeoJSONGeometryCollection2
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPoint4 {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString4 {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon4 {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint4 {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString4 {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon4 {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONGeometryCollection2 {
  type: "GeometryCollection"
  geometries: (
    | GeoJSONPoint5
    | GeoJSONLineString5
    | GeoJSONPolygon5
    | GeoJSONMultiPoint5
    | GeoJSONMultiLineString5
    | GeoJSONMultiPolygon5
  )[]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPoint5 {
  type: "Point"
  coordinates: [number, number, ...number[]]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONLineString5 {
  type: "LineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONPolygon5 {
  type: "Polygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPoint5 {
  type: "MultiPoint"
  coordinates: [number, number, ...number[]][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiLineString5 {
  type: "MultiLineString"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
export interface GeoJSONMultiPolygon5 {
  type: "MultiPolygon"
  coordinates: [
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    [number, number, ...number[]],
    ...[number, number, ...number[]][]
  ][][]
  bbox?: [number, number, number, number, ...number[]]
  [k: string]: unknown
}
```

Reply:
La définition chez nous des IFeature, IGeometry et IProperties colle au retour effectif de l'api https://api-adresse.data.gouv.fr/search?q= . On en reparle lundi pour que je comprenne bien ce qui te pose souci.
J'ai souhaité récupéré la feature plutôt que le geopoint car j'ai besoin de plus que les coordonnées spatiales. La feature me donne aussi city, postcode et name

Reply:
La définition actuelle dans le code ne correspond pas à ce qui est effectivement renvoyé. On a parfois des données qui ne sont pas de type `Point`